src/main/java/com/redhat/devtools/lsp4ij/features/selectionRange/AbstractLSPExtendWordSelectionHandler.java

```diff
@@ -0,0 +1,50 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+
+package com.redhat.devtools.lsp4ij.features.selectionRange;
+
+import com.intellij.codeInsight.editorActions.ExtendWordSelectionHandler;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.redhat.devtools.lsp4ij.LanguageServersRegistry;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Abstract base class for 'extendWordSelectionHandler' implementations for LSP4IJ files.
+ */
+abstract class AbstractLSPExtendWordSelectionHandler implements ExtendWordSelectionHandler {
```

Comment:
Added a new common base class for all LSP `extendWordSelectionHandler` implementations that just makes sure that the file look proper and is supported by a configured language server.